### PR TITLE
fix: add has polyfill for alignment

### LINKED ISSUE #2704: Hero block - `.stk-container:has(>.stk--column-flex)` is missing for Firefox browser
**Describe the bug**
- Issue for Firefox browser only
- Fix might affect other blocks

**To Reproduce**
1. Add a new page
2. Add a Hero block
3. Save page
4. Go to frontend
5. Right click and click Inspect
6. In Styles, search "has" in Filter field
7. See all results and check if the class mentioned above is visible (for Chrome)
8. Repeat steps 4-7 for Firefox
9. See that the class is missing

**Expected behavior**
- See screenshot (Google Chrome): the class is visible

https://github.com/gambitph/Stackable/assets/103395655/24361081-f50c-409c-bf32-f08af8d7a1f6

**Screenshots**
- Firefox: the class is missing

https://github.com/gambitph/Stackable/assets/103395655/68a843ec-7708-4f9f-8e1a-6fc3624a6316

**Desktop:**
 - Browser: Firefox
 - Browser Version: 114.0.1 (64-bit)

### LINKED ISSUE #2784: Alignment: Add compatibility for browsers that do not support :has
The CSS in here uses `:has`, this pseudo selector isn't supported in Firefox and Safari <= 15.3:

https://github.com/gambitph/Stackable/blob/ebc09fa4b8a46e8b0cc1931e06353dffb7628e52/src/block-components/alignment/style.scss#L8-L11

I suggest to create some JavaScript for the frontend that will add a custom class name to the element so that the necessary style are included, and new CSS rules to support it. The JS file should only be loaded if the browser is Firefox or Safari <= 15.3

For loading the JS conditionally depending on the browser, the user agent can be used to check similar to https://github.com/gambitph/Stackable/blob/7a7a556f46dea10a98eec8b4dbfe4f088bd0ea8d/src/block/accordion/index.php#L41

In the JS, be sure to support dynamically added DOM elements, use MutationObserver to check for this.

Create a JS file `src/block-components/alignment/frontend-has-polyfill.js` and add your JS polyfill there. Then to compile the script into the `dist/` folder, refer to:
- https://github.com/gambitph/Stackable/blob/develop/.config/webpack.config.dev.js#L172
- https://github.com/gambitph/Stackable/blob/develop/.config/webpack.config.prod.js#L157

### PR DESCRIPTION
**Pull request by mxkae**
fixes #2784, fixes #2704

### CODE REVIEW

File: src/block-components/alignment/frontend-has-polyfill.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * WordPress dependencies
+ */
+import domReady from '@wordpress/dom-ready'
+
+class StackableAlignmentPolyfill {
+	init = () => {
+		const modifyCSS = () => {
+			const containers = document.querySelectorAll( '.stk-container' )
+			containers.forEach( container => {
+				const columnFlex = container.querySelector( '.stk--column-flex' )
```
Comment (bfintal) [inline src/block-components/alignment/frontend-has-polyfill.js:11]: Based on the selector we're trying to mimic, we need this to be for the direct descendent
```suggestion
				const columnFlex = container.querySelector( ':scope > .stk--column-flex' )
```

File: src/block-components/alignment/frontend-has-polyfill.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+				if ( columnFlex ) {
+					container.classList.add( 'stk-container--has-child-column-flex' )
+				} else if ( ! columnFlex && container.classList.contains( 'stk-container--has-child-column-flex' ) ) {
+					container.classList.remove( 'stk-container--has-child-column-flex' )
```
Comment (bfintal) [inline src/block-components/alignment/frontend-has-polyfill.js:15]: Do you still need this?

**Review by bfintal — CHANGES_REQUESTED** [review]
I took the liberty of changing some file/script/class names to better reflect the purpose of the newly introduced script.

I have some remaining suggestions / improvements:

1. We can have better names for the new classes that can signify that they are polyfill classes:
- `stk--height-100` ➡️ `stk--height-100-polyfill`
- `stk-container--has-child-column-flex` ➡️ `stk-container--has-child-column-flex-polyfill`

2. Bring back the specificity of the original selectors you're trying to polyfill, because if not the styles might not be applied correctly.

You're mainly trying to add a class that would have the same effect as the following selectors. But these 2 sets of selectors have different specificities (you can check here: https://specificity.keegan.st/)

- `:is(.stk-block-content, .stk-inner-blocks):not(.stk--column-flex):has(> :is(.stk--block-margin-top-auto, .stk--block-margin-bottom-auto))` ➡️ 3
- `.stk-container:has(> .stk--column-flex)` ➡️ 2

The new selectors are just:

- `.stk--height-100` ➡️ 1
- `.stk-container--has-child-column-flex` ➡️ 1

That means the styles might get easily overridden by other selectors. You'll need to mimic / bump up the specificity to the same level.

**Comment by andeng1106** [comment]
@mxkae @bfintal Fixes are not working in editor for Firefox browser only

some features & scenarios tried:
* innerBlockJustify setting
* innerBlockAlign setting (Inner Column)
* in the inner column block, align all blocks to the left, then the last child block make margin left auto - the last child block should go to the right

**Review by bfintal — CHANGES_REQUESTED** [review]
I feel it's odd that `Chrome/` is part of the detection for Safari. It should be `Safari/` and `Version/` and the number after version should be <= `15.3`

File: src/block-components/alignment/index.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		$user_agent = ! empty( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '';
+
+		if ( ! $user_agent ) {
+			return;
+		}
+
+		$load_polyfill = false;
+
+		// Safari <= 15.3
+		if ( stripos( $user_agent, 'Version/' ) && stripos( $user_agent, 'Safari/' ) !== false ) {
```
Comment (bfintal) [inline src/block-components/alignment/index.php:24]:
```suggestion
		if ( stripos( $user_agent, 'Version/' ) !== false && stripos( $user_agent, 'Safari/' ) !== false ) {
```

**Pull request merged**